ciao-controller/instance.go

```diff
@@ -107,14 +108,20 @@ func newInstance(ctl *controller, tenantID string, workload *types.Workload,
 func (i *instance) Add() error {
 	ds := i.ctl.ds
 	var err error
-	if i.CNCI == false {
-		err = ds.AddInstance(&i.Instance)
-	} else {
-		err = ds.AddTenantCNCI(i.TenantID, i.ID, i.MACAddress)
-	}
+	err = ds.AddInstance(&i.Instance)
 	if err != nil {
 		return errors.Wrapf(err, "Error creating instance in datastore")
 	}
+
+	if i.CNCI == true {
+		t, err := ds.GetTenant(i.TenantID)
+		if err != nil {
+			return errors.Wrap(err, "Error creating instance")
```

Comment:
the result of the err is that there is an error creating an instance. I will make the message more verbose.